File: ios/Talise/Features/Onboarding/WelcomeView.swift
import SwiftUI

/// Hero onboarding screen — first thing a fresh install sees after the
/// splash. Mossy-green radial gradient washes the top half down into
/// black; the Talise pinwheel sits in the upper-middle; bottom-left
/// headline + subtitle copy frames two CTAs (primary "Get Started" and
/// secondary "I have an account" for returning users) over a tiny
/// Terms acknowledgement footer.
///
/// `onContinue` → start the brand-intro carousel (new user path).
/// `onSignIn`   → jump straight to the sign-in sheet (returning user).
struct WelcomeView: View {
    let onContinue: () -> Void
    let onSignIn: () -> Void

    var body: some View {
        ZStack {
            // Black base so the gradient's lower half blends cleanly
            // into the safe-area.
            TaliseColor.bg.ignoresSafeArea()

            // Mossy green radial wash anchored at the top. The center
            // sits just above the screen top so the brightest area is
            // at the device's top edge, falling off to black around
            // ~55% of the viewport height. The hex (0x6BA85A) is a
            // softened forest green — the regular accent (0x79D96C)
            // is too vivid for a full-width wash.
            GeometryReader { proxy in
                RadialGradient(
                    colors: [
                        Color(hex: 0x6BA85A).opacity(0.95),
                        Color(hex: 0x2A3E22).opacity(0.6),
                        Color.black.opacity(0.0),
                    ],
                    center: UnitPoint(x: 0.5, y: -0.05),
                    startRadius: 0,
                    endRadius: proxy.size.height * 0.55
                )
                .ignoresSafeArea()
            }

            VStack(spacing: 0) {
                Spacer().frame(height: 120)

                logoMark
                    .frame(width: 96, height: 96)

                Spacer()

                copyBlock
                    .padding(.horizontal, 24)
                    .padding(.bottom, 20)

                primaryCTA
                    .padding(.horizontal, 24)
                    .padding(.bottom, 12)

                secondaryCTA
                    .padding(.horizontal, 24)
                    .padding(.bottom, 16)

                termsFooter
                    .padding(.horizontal, 32)
                    .padding(.bottom, 24)
            }
        }
        .preferredColorScheme(.dark)
    }

    // ── Subviews ────────────────────────────────────────────────────

    @ViewBuilder
    private var logoMark: some View {
        if UIImage(named: "TaliseLogo") != nil {
            Image("TaliseLogo")
                .resizable()
                .scaledToFit()
        } else {
            Pinwheel()
        }
    }

    private var copyBlock: some View {
        VStack(alignment: .leading, spacing: 10) {
            Text("Move money without borders")
                .font(TaliseFont.heading(26, weight: .medium))
                .kerning(-0.6)
                .foregroundStyle(TaliseColor.fg)
                .multilineTextAlignment(.leading)

            Text(
                "Moving money across the world is complex, Talise brings simplicity to this. Free transactions, smart money movement."
            )
            .font(TaliseFont.body(13, weight: .light))
            .foregroundStyle(TaliseColor.fgMuted)
            .multilineTextAlignment(.leading)
            .lineSpacing(2)
            .fixedSize(horizontal: false, vertical: true)
        }
        .frame(maxWidth: .infinity, alignment: .leading)
    }

    private var primaryCTA: some View {
        Button(action: onContinue) {
            Text("Get Started")
                .font(TaliseFont.body(15, weight: .medium))
                .foregroundStyle(TaliseColor.bg)
                .frame(maxWidth: .infinity)
                .frame(height: 54)
                .background(TaliseColor.fg)
                .clipShape(Capsule())
        }
        .buttonStyle(.plain)
    }

    private var secondaryCTA: some View {
        Button(action: onSignIn) {
            Text("I have an account")
                .font(TaliseFont.body(15, weight: .medium))
                .foregroundStyle(TaliseColor.fg)
                .frame(maxWidth: .infinity)
                .frame(height: 54)
                .background(TaliseColor.surface2)
                .clipShape(Capsule())
        }
        .buttonStyle(.plain)
    }

    private var termsFooter: some View {
        // Light footer ack — `Terms and Conditions` is underlined so it
        // reads as a link target. No tap handler yet (we don't have a
        // terms page route); when one ships, wrap the Text in a Button
        // that opens it.
        (Text("You accept ")
            + Text("Terms and Conditions").underline()
            + Text(" by continuing."))
            .font(TaliseFont.body(11, weight: .light))
            .foregroundStyle(TaliseColor.fgDim)
            .multilineTextAlignment(.center)
            .frame(maxWidth: .infinity)
    }
}

/// Hand-drawn approximation of the white pinwheel brand mark — same
/// geometry as `HomeView`'s `TaliseLogoMark`, duplicated here so the
/// Onboarding feature stays self-contained.
private struct Pinwheel: View {
    var body: some View {
        Canvas { ctx, size in
            let cx = size.width / 2
            let cy = size.height / 2
            let r: CGFloat = size.width * 0.22
            for i in 0..<4 {
                let angle = CGFloat(i) * .pi / 2
                var transform = CGAffineTransform(translationX: cx, y: cy)
                transform = transform.rotated(by: angle)
                transform = transform.translatedBy(x: 0, y: -size.height * 0.28)
                let rect = CGRect(
                    x: -r * 0.45, y: -r * 0.55,
                    width: r * 0.9, height: r * 1.15
                ).applying(transform)
                let path = Path(ellipseIn: rect)
                ctx.fill(path, with: .color(.white))
            }
        }
    }
}

File: android/app/src/main/java/io/talise/app/feature/onboarding/WelcomeView.kt
package io.talise.app.feature.onboarding

import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.BoxWithConstraints
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.navigationBarsPadding
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.SpanStyle
import androidx.compose.ui.text.buildAnnotatedString
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.style.TextDecoration
import androidx.compose.ui.text.withStyle
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import io.talise.app.R
import io.talise.app.ui.theme.TaliseColors
import io.talise.app.ui.theme.TaliseType

/**
 * Hero onboarding screen — iOS `WelcomeView`. A mossy-green wash fills the top ~42% of
 * the viewport then falls hard into black; the Talise mark sits centered just below the
 * gradient transition; bottom-left "Move money without borders" headline + supporting
 * subtitle frames the two CTAs (primary "Get Started", secondary "I have an account")
 * above a small Terms acknowledgement footer.
 *
 * `onContinue` → new user path. `onSignIn` → straight to sign-in (returning user).
 */
@Composable
fun WelcomeView(onContinue: () -> Unit, onSignIn: () -> Unit) {
    BoxWithConstraints(Modifier.fillMaxSize().background(TaliseColors.bg)) {
        val screenH = maxHeight

        // Top green wash — linear, even across the width, fading vertically. Stops
        // match iOS: 0% mossy green, 45% transition midpoint, 85% pure black. Sized
        // to ~42% of the screen height so the logo lands just below it.
        Box(
            Modifier
                .fillMaxWidth()
                .height(screenH * 0.42f)
                .background(
                    Brush.verticalGradient(
                        0.0f to Color(0xFF6BA85A),
                        0.45f to Color(0xFF355626),
                        0.85f to Color(0xFF000000),
                        1.0f to Color(0xFF000000),
                    )
                )
        )

        Column(
            Modifier.fillMaxSize().navigationBarsPadding(),
            horizontalAlignment = Alignment.CenterHorizontally,
        ) {
            // Logo positioned just below the gradient's end (~50% of screen height).
            Spacer(Modifier.height(screenH * 0.42f))

            Image(
                painter = painterResource(R.drawable.taliselogo),
                contentDescription = "Talise",
                contentScale = ContentScale.Fit,
                modifier = Modifier.size(88.dp),
            )

            Spacer(Modifier.weight(1f))

            CopyBlock(
                Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 20.dp)
                    .padding(bottom = 22.dp)
            )

            PrimaryCta(
                onClick = onContinue,
                modifier = Modifier
                    .padding(horizontal = 24.dp)
                    .padding(bottom = 12.dp),
            )

            SecondaryCta(
                onClick = onSignIn,
                modifier = Modifier
                    .padding(horizontal = 24.dp)
                    .padding(bottom = 18.dp),
            )

            TermsFooter(
                Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 32.dp)
                    .padding(bottom = 28.dp)
            )
        }
    }
}

@Composable
private fun CopyBlock(modifier: Modifier = Modifier) {
    Column(modifier) {
        // Headline — Figma spec: 23.5px / 600 / -0.705 letter spacing.
        Text(
            "Move money without borders",
            style = TaliseType.heading(23.5.sp, FontWeight.SemiBold),
            letterSpacing = (-0.705).sp,
            color = TaliseColors.fg,
            maxLines = 1,
        )
        Spacer(Modifier.height(10.dp))
        Text(
            "Moving money across the world is complex, Talise brings simplicity to this. No network fees, smart money movement.",
            style = TaliseType.body(13.sp, FontWeight.Light),
            letterSpacing = (-0.39).sp,
            lineHeight = 17.sp,
            color = TaliseColors.fgMuted,
        )
    }
}

@Composable
private fun PrimaryCta(onClick: () -> Unit, modifier: Modifier = Modifier) {
    Box(
        modifier = modifier
            .fillMaxWidth()
            .height(54.dp)
            .clip(RoundedCornerShape(27.dp))
            .background(TaliseColors.fg)
            .clickable(onClick = onClick),
        contentAlignment = Alignment.Center,
    ) {
        Text(
            "Get Started",
            style = TaliseType.body(15.sp, FontWeight.Medium),
            letterSpacing = (-0.45).sp,
            color = TaliseColors.bg,
        )
    }
}

@Composable
private fun SecondaryCta(onClick: () -> Unit, modifier: Modifier = Modifier) {
    // Glassmorphic capsule — soft white fill + specular hairline so the pill reads
    // as a frosted surface against the black page (iOS `taliseGlass(cornerRadius: 27)`).
    Box(
        modifier = modifier
            .fillMaxWidth()
            .height(54.dp)
            .clip(RoundedCornerShape(27.dp))
            .background(Color.White.copy(alpha = 0.08f))
            .border(1.dp, Color.White.copy(alpha = 0.12f), RoundedCornerShape(27.dp))
            .clickable(onClick = onClick),
        contentAlignment = Alignment.Center,
    ) {
        Text(
            "I have an account",
            style = TaliseType.body(15.sp, FontWeight.Medium),
            letterSpacing = (-0.45).sp,
            color = TaliseColors.fg,
        )
    }
}

@Composable
private fun TermsFooter(modifier: Modifier = Modifier) {
    Text(
        buildAnnotatedString {
            append("You accept ")
            withStyle(SpanStyle(textDecoration = TextDecoration.Underline)) {
                append("Terms and Conditions")
            }
            append(" by continuing.")
        },
        style = TaliseType.body(11.sp, FontWeight.Light),
        letterSpacing = (-0.33).sp,
        color = TaliseColors.fgDim,
        textAlign = TextAlign.Center,
        modifier = modifier,
    )
}
